jobs/uaa-customized/templates/web/layouts/main.html

```diff
@@ -1,20 +1,17 @@
 <!DOCTYPE html>
-<html xmlns:th="http://www.thymeleaf.org"
-      xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout"
-      xmlns:sec="http://www.thymeleaf.org/thymeleaf-extras-springsecurity3"
+<html xmlns:th="http://www.thymeleaf.org" xmlns:layout="http://www.ultraq.net.nz/thymeleaf/layout" dir="ltr" lang="en"
       th:with="assetBaseUrl=${@environment.getProperty('assetBaseUrl','/resources/oss')},
-      isUaa=${T(org.cloudfoundry.identity.uaa.zone.IdentityZoneHolder).isUaa()},
-      zoneName=${T(org.cloudfoundry.identity.uaa.zone.IdentityZoneHolder).get().getName()},
-      branding=${T(org.cloudfoundry.identity.uaa.zone.MergedZoneBrandingInformation).resolveBranding()},
-      companyName=${branding.getCompanyName() ?: 'Cloud Foundry'},
-      copyrightCompanyName=${branding.getCompanyName() ?: 'CloudFoundry.org Foundation, Inc. '},
-      copyrightFallback=${'Copyright &#169; ' + copyrightCompanyName + ' ' + #dates.year(#dates.createNow()) + '. All Rights Reserved.'},
-      copyright=${branding.getFooterLegalText() ?: copyrightFallback},
-      brandingLogo=${branding.getProductLogo()},
-      fallbackLogo=${isUaa ? assetBaseUrl+'/images/product-logo.png' : null},
-      logo=${brandingLogo != null ? 'data:image/png;base64,'+#strings.replace(brandingLogo,T(java.lang.System).getProperty('line.separator'),'') : fallbackLogo}"
-      dir="ltr"
-      lang="en"
+                isUaa=${T(org.cloudfoundry.identity.uaa.zone.IdentityZoneHolder).isUaa()},
+                zoneName=${T(org.cloudfoundry.identity.uaa.zone.IdentityZoneHolder).get().getName()},
+                branding=${T(org.cloudfoundry.identity.uaa.zone.MergedZoneBrandingInformation).resolveBranding()},
+                companyName=${branding.getCompanyName() ?: 'Cloud Foundry'},
+                copyrightCompanyName=${branding.getCompanyName() ?: 'CloudFoundry.org Foundation, Inc. '},
+                copyrightFallback=${'Copyright &#169; ' + copyrightCompanyName + ' ' + #dates.year(#dates.createNow()) + '. All Rights Reserved.'},
+                copyright=${branding.getFooterLegalText() ?: copyrightFallback},
+                brandingLogo=${branding.getProductLogo()},
+                fallbackLogo=${isUaa ? assetBaseUrl+'/images/product-logo.png' : null},
+                urlParamError=${param.error},
```

Comment:
Does it work if you do `urlParamError=${#conversions.convert(param.error, 'java.lang.Boolean')},` ?

Rationale: if we coerce the value to a boolean **here**, it makes it _impossible_ to accidentally abuse `urlParamError` elsewhere in a way that would allow arbitrary injection.

Reply:
Don't mind me, just going to find out if this works on top of your branch...

Reply:
Answered my own question: "no, it doesn't work".